torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2990,6 +2990,117 @@ def generator():
 
     return list(generator())
 
+def sample_inputs_max_unpool1d(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    kerneli = [(1,), 3]
+    stridei = [None, (2,)]
+    Ni = [1]
+    Ci = [2]
+    Wi = [6]
+    ceil_modei = [True, False]
+    paddingi = [0]
+    dilationi = [1]
+    output_sizei = [True, False]
+
+    products = product(kerneli, stridei, Ni, Ci, Wi, ceil_modei, paddingi, dilationi, output_sizei)
+
+    def generator():
+        for kernel_size, stride, N, C, W, ceil_mode, padding, dilation, output_size in products:
+            input_tensor = make_arg((N, C, W))
+
+            max_unpool1d = torch.nn.MaxUnpool1d(kernel_size, stride, padding)
+            max_pool1d = torch.nn.MaxPool1d(kernel_size, stride, ceil_mode=ceil_mode, padding=padding,
+                                            dilation=dilation, return_indices=True)
+
+            kwargs = {
+                "kernel_size": max_unpool1d.kernel_size,
```

Comment:
Is there a reason we need to declare `max_unpool1d` and get these kwargs back out from it instead of using them directly?

Reply:
Thanks for the question, @saketh-are. Using `nn.MaxUnpool1d` allows to test default behavior of these kwargs, and what happens when passed `None`. But for this PR, I've removed OpInfo changes since these were incorporated in a different PR by @nikitaved. (I'll link to the PR soon)

Reply:
Update: looks like I was wrong. The OpInfos for `max_unpool{}d` aren't there yet. @nikitaved - Do you plan to work on them? Or can I add the OpInfos in this PR? Please let me know if I missed any open PR on this. (same question for `max_pool{}d_with_indices`)

Reply:
I think it makes sense to keep OpInfo tests from https://github.com/pytorch/pytorch/pull/68625 as they do not skip gradchecks. Waiting on @mruberry to approve them.

Reply:
Ah, got it. Sorry, I missed it. Thanks, @nikitaved. Quick Question: are you working on OpInfos for `max_pool{}d_with_indices` as well?

Reply:
Not yet, no.